controllers/storageclassclaimsecret/storageclassclaimsecret_controller.go

```diff
@@ -0,0 +1,665 @@
+		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
+		r.cephBlockPool.Namespace = r.OperatorNamespace
+	} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
+		r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s", r.storageConsumer.Name)
+		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
+	}
+
+	r.cephClientProvisioner = &rookCephv1.CephClient{}
+	r.cephClientProvisioner.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "provisioner")
+	r.cephClientProvisioner.Namespace = r.OperatorNamespace
+
+	r.cephClientNode = &rookCephv1.CephClient{}
+	r.cephClientNode.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "node")
+	r.cephClientNode.Namespace = r.OperatorNamespace
+
+	r.cephResourcesByName = map[string]*v1alpha1.CephResourcesSpec{}
+
+	for _, cephResourceSpec := range r.storageClassClaim.Status.CephResources {
+		r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+	}
+
+	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimCreating
+
+	if r.storageClassClaimSecret.GetDeletionTimestamp().IsZero() {
+		if r.storageClassClaim.Spec.Type == "blockpool" {
+
+			if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientRBDNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephBlockPool(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+		} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+			if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientCephFSNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		cephResourcesReady := true
+		for _, cephResource := range r.storageClassClaim.Status.CephResources {
+			if cephResource.Phase != "Ready" {
+				cephResourcesReady = false
+				break
+			}
+		}
+
+		if cephResourcesReady {
+			r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
+		}
+
+	} else {
+		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *StorageClassClaimSecretReconciler) reconcileCephBlockPool() error {
+
+	failureDomain := r.storageCluster.Status.FailureDomain
+
+	capacity := r.storageConsumer.Spec.Capacity.String()
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephBlockPool, func() error {
+		if err := r.own(r.cephBlockPool); err != nil {
+			return err
+		}
+		deviceSetList := r.storageCluster.Spec.StorageDeviceSets
+		var deviceSet *v1.StorageDeviceSet
+		for i := range deviceSetList {
+			ds := &deviceSetList[i]
+			if ds.Name == "default" {
```

Comment:
Looks like this is old code. Do we have a plan to include the IOPS feature in this controller?

Reply:
By technicality, yes, since it needs to maintain parity with whatever the StorageClassClaim controller is doing on the provider side.

Reply:
@jarrpa we already merged the 4.12 SCC changes for the provider side on `main`. I would suggest to rebase ASAP

Reply:
We'll see if a rebase is even feasible, since these are completely new files that have been created as copies. Even so it's not impossible.

Reply:
The rebase itself wasn't too bad, but things immediately broke afterwards. I managed to find and resolve the two small bugs from #1750, then cherry-picked Madhu's fixes until I could do another rebase earlier today. This should now be resolved.